### ISSUE TITLE
Feature request: add Auto Redirect option

### ISSUE THREAD

**Issue by Jpxe** [libredirect/libredirect#89]
Hello I just discovered this project (I've been using Privacy Redirect before) and this seems great!
I wanted to ask if it would be possible to add [Farside](https://github.com/benbusby/farside) support?

**Comment by ManeraKai** [comment]
> Hello I just discovered this project (I've been using Privacy Redirect before) and this seems great!

I've seen you help Simon with OSM, that issue helped me a lot with implementing FacilMap. It's still not finished but working on it.

LibRedirect is Farside but everything is locally done. Even most of the instances are updated automatically.

**Comment by Jpxe** [comment]
> I've seen you help @SimonBrazell with OSM, that issue helped me a lot with implementing FacilMap. It's still not finished but working on it.

That's great to hear, I'm glad I could be of help.

> LibRedirect is Farside but everything is locally done. Even most of the instances' lists are updated automatically.

Interesting, as I understand it Farside detects if an instance is not working and automatically redirects to a working one, does LibRedirect work in the same way?

**Comment by ManeraKai** [comment]
> as I understand it Farside detects if an instance is not working and automatically redirects to a working one, does LibRedirect work in the same way?

Before LibRedirect used to be like that, but because of it being automatic, it went off crazy sometimes and I wasn't really comfortable with the idea of an extension hopping between many intsance so fast and crashing the browser. I don't want the extension to get out of the user's hand since it's designed to control requests and preserve your privacy.

So, if you faced an offline instance, just open the popup and click change instance. This is a lot better imo.

**Comment by Jpxe** [comment]
I get that, the way LibRedirect behaves now by default, with manual refresh, definitely makes sense.

However as an option, is there a way I can add Farside as a custom instance? I would like to try it out myself and it seems LibRedirect removes the "/" character in the URL and everything after it. For example `https://farside.link/bibliogram/` becomes `https://farside.link/` making it unusable.

**Comment by ManeraKai** [comment]
What will you use Farside for?

**Comment by Jpxe** [comment]
I want to test it out to see if it could improve my experience using some services with lots of not working instances. I've been having a lot of problems with bibliogram and some times invidious in the past.

**Comment by ManeraKai** [comment]
For bibliogram, even the official main isntance is offline: https://bibliogram.art Update the list and there won't be a difference between this and farside aside from automatic redirection of offline instances.

**Comment by ManeraKai** [comment]
Do you want an option for automatic redireciton?

**Comment by Jpxe** [comment]
Yes that would be great

**Comment by ManeraKai** [comment]
Forgot to mention the issue number in the commit: https://github.com/libredirect/libredirect/commit/04fcac392067c28a875648bcc3ac234d6ed05b05
Although it's automatic, to prevent any browser crashing due to an infinite loop, you'll be redirect to a page saying you will be redirected after 2 seconds, and then you'll be redirect.

**Comment by Jpxe** [comment]
Thanks for adding this! 

I'm having some minor issues with what seems like, unnecessary redirects, I'll try test it some more with a clean browser install and get back to you.

**Comment by kartikynwa** [comment]
Sorry for necrobumping. I was thinking being able to use farside as an instance would be a good option to have alongside the inbuilt redirection. How this would work can be pretty straighforward. For twitter for example, one can add a custom instance with the URL "https://farside.link/nitter". But when I tried this, the problem I faced was that upon adding "https://farside.link/nitter", libredirect is discarding "/nitter" from it so it ends up not being working. Thoughts on this?
